crates/router/src/connector/netcetera/transformers.rs

```diff
@@ -283,3 +370,265 @@ impl TryFrom<&NetceteraRouterData<&types::authentication::PreAuthNRouterData>>
+            now,
+            common_utils::date_time::DateFormat::YYYYMMDDHHmm,
+        )
+        .change_context(errors::ConnectorError::RequestEncodingFailedWithReason(
+            "Failed to format Date".to_string(),
+        ))?;
+        let request = item.router_data.request.clone();
+        let pre_authn_data = request.pre_authentication_data.clone();
+        let three_ds_requestor = netcetera_types::ThreeDSRequestor {
+            three_ds_requestor_authentication_ind:
+                netcetera_types::ThreeDSRequestorAuthenticationIndicator::Payment,
+            three_ds_requestor_authentication_info: Some(
+                netcetera_types::SingleOrListElement::new_single(
+                    netcetera_types::ThreeDSRequestorAuthenticationInformation {
+                        three_ds_req_auth_method: "01".to_string(),
+                        three_ds_req_auth_timestamp,
+                        three_ds_req_auth_data: None,
+                    },
+                ),
+            ),
+            three_ds_requestor_challenge_ind: None,
+            three_ds_requestor_prior_authentication_info: None,
+            three_ds_requestor_dec_req_ind: None,
+            three_ds_requestor_dec_max_time: None,
+            app_ip: None,
+            three_ds_requestor_spc_support: None,
+            spc_incomp_ind: None,
+        };
+        let card = utils::get_card_details(request.payment_method_data, "netcetera")?;
+        let cardholder_account = netcetera_types::CardholderAccount {
+            acct_type: None,
+            card_expiry_date: Some(card.get_expiry_date_as_yymm()?),
+            acct_info: None,
+            acct_number: card.card_number,
+            scheme_id: None,
+            acct_id: None,
+            pay_token_ind: None,
+            pay_token_info: None,
+            card_security_code: Some(card.card_cvc),
+        };
+        let currency = request
+            .currency
+            .get_required_value("currency")
+            .change_context(errors::ConnectorError::MissingRequiredField {
+                field_name: "currency",
+            })?;
+        let purchase = netcetera_types::Purchase {
+            purchase_instal_data: None,
+            merchant_risk_indicator: None,
+            purchase_amount: request.amount,
+            purchase_currency: currency.iso_4217().to_string(),
+            purchase_exponent: currency.number_of_digits_after_decimal_point(),
+            purchase_date: Some(
+                common_utils::date_time::format_date(
+                    now,
+                    common_utils::date_time::DateFormat::YYYYMMDDHHmmss,
+                )
+                .change_context(
+                    errors::ConnectorError::RequestEncodingFailedWithReason(
+                        "Failed to format Date".to_string(),
+                    ),
+                )?,
+            ),
+            recurring_expiry: Some("20240401".to_string()),
```

Comment:
Why is this hardcoded?

Reply:
This is a temporary solution. Core is not ending these values currently but netcetera requires these fields to be sent. Core changes to send these info in `router_data.request` will be done in near future.